compiler/rustc_mir_transform/src/uninhabited_enum_branching.rs

```diff
@@ -71,6 +72,29 @@ fn variant_discriminants<'tcx>(
     }
 }
 
+/// Ensures that the `otherwise` branch leads to an unreachable bb, returning `None` if so and a new
+/// bb to use as the new target if not.
+fn ensure_otherwise_unreachable<'tcx>(
+    body: &Body<'tcx>,
+    targets: &SwitchTargets,
+) -> Option<BasicBlockData<'tcx>> {
+    let otherwise = targets.otherwise();
+    let bb = &body.basic_blocks()[otherwise];
+    if bb.terminator().kind == TerminatorKind::Unreachable
+        && !bb.is_cleanup
```

Comment:
`!bb.is_cleanup` condition should be unnecessary.

Reply:
Yeah, I was not aware of the exact semantics of this previously, so I had put this in to be safe.